pkg/util/pki/generate.go

```diff
@@ -157,6 +158,65 @@ func EncodeECPrivateKey(pk *ecdsa.PrivateKey) ([]byte, error) {
 	return pem.EncodeToMemory(block), nil
 }
 
+// TranslateToKeyEncoding translates keyBytes, a byte slice containing PEM formatted private key
+// and translates it to the format as specified by keyEncoding, i.e. PKCS#1 or PKCS#8.
+// If keyBytes already in desired format, it returns the same slice.
+func TranslateToKeyEncoding(keyBytes []byte, keyEncoding v1alpha2.KeyEncoding) ([]byte, error) {
```

Comment:
This method shouldn't be necessary if we switch to using `EncodePrivateKey` instead 😄